compiler-core/src/error.rs

```diff
@@ -1881,31 +1882,57 @@ assigned variables to all of them."
                     }
                 }
 
-                TypeError::UpdateMultiConstructorType { location } => {
-                    let text = wrap("This type has multiple constructors \
-so it cannot be safely updated. If this value was one of the other variants \
-then the update would be produce incorrect results.
-
-Consider pattern matching on it with a case expression and then\
-constructing a new record with its values."
-                );
-
-                    Diagnostic {
-                        title: "Unsafe record update".into(),
-                        text,
-                        hint: None,
-                        level: Level::Error,
-                        location: Some(Location {
-                            label: Label {
-                                text: Some("I can't tell this is always the right constructor".into()),
-                                span: *location,
-                            },
-                            path: path.clone(),
-                            src: src.clone(),
-                            extra_labels: vec![],
-                        }),
+                TypeError::UnsafeRecordUpdate { location, reason } =>
+                    match reason {
+                        UnsafeRecordUpdateReason::UnknownVariant {constructed_variant} => {
+                            let text = wrap_format!("
+I cannot use this value to build an updated `{constructed_variant}` as it could be
```

Comment:
Remove this hard-coded newline by adding back the \ at the end please

Reply:
Done!